app/Modules/Core/Model/EventModel.php

```diff
@@ -154,4 +154,29 @@ public function findPreviousEvent(int $eventSeriesId): ?IRow
             ->order('end DESC')
             ->fetch() ?: null;
     }
+
+    /**
+     * @return Event[]
+     */
+    public function getApprovedEventsByDate(
```

Comment:
Do you plan to use this method in the near future? How about make it simple as you need. No `$to` or `$orderByRate`?
The usage then would be way simpler.